backend/src/main/java/com/google/step/servlets/AuthenticationServlet.java

```diff
@@ -0,0 +1,70 @@
+package com.google.sps.servlets;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.users.User;
+import com.google.gson.Gson;
+
+@WebServlet("/authentication")
+public class AuthenticationServlet extends HttpServlet {
+    /**
+     * Checks state of user in this class.
+     */
+    private static UserService userService;
+
+    @Override
+    public void init() {
+        userService = UserServiceFactory.getUserService();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String url;
+        boolean loggedIn = userService.isUserLoggedIn();
+        if(loggedIn){
+            url = userService.createLogoutURL("/");
+        }
```

Comment:
nit: put the else with the previous "}" and put a space between. (i.e. } else { ).

Reply:
fixed